### ISSUE TITLE
yolov8 model architecture

### ISSUE THREAD

**Issue by Christian-lyc** [ultralytics/ultralytics#16501]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, I want to ask if there 's any solution for providing a model architecture code instead of yaml? The current saved pt model file can not be loaded to my python code constructed model. Thank you

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Christian-lyc, thank you for reaching out to Ultralytics 🚀!

To get started, we recommend reviewing our [Docs](https://docs.ultralytics.com) for detailed information on various topics, including using YOLOv8 models in your Python projects. You might find our [Python usage examples](https://docs.ultralytics.com/usage/python/) especially helpful.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us investigate the issue further.

For questions related to custom usage ❓, please include more details about your use case, any error messages you are encountering, and your current setup. Also, ensure you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time support, visit [Discord](https://ultralytics.com/discord) 🎧. Prefer forums? Check out [Discourse](https://community.ultralytics.com) or our [Subreddit](https://reddit.com/r/ultralytics) for discussions.

## Upgrade

Ensure you have the latest version of the `ultralytics` package installed along with all required dependencies in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

Run YOLOv8 in various verified environments to ensure full compatibility, including:

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

This badge shows the status of our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule). If it's green, all tests across different platforms are passing successfully.

Please note, this is an automated response. An Ultralytics engineer will review your issue and provide further assistance soon. 😊

**Comment by Y-T-G** [comment]
If you want to load the model outside ultralytics, you can export it to a different format.

https://docs.ultralytics.com/modes/export/

**Comment by Christian-lyc** [comment]
Thank you so much for the quick reply. Actually,  I want to use structure pruning and remove the zero-out channels. But Ultralytics model is written on yaml. So I tried to build a new yolov8n model in a python script way. But there's a mismatch of key words of state_dict. Since I'm not familiar with how to build model on yaml file. I want to ask if there's also an official  python script way to build the model, that will help me to load the best.pt model. Thank you so much

**Comment by Y-T-G** [comment]
The yaml just defines the structure of the model. The model is created based on the yaml using PyTorch.

You can see all the modules by running `print(model.model.model)`.

**Comment by Christian-lyc** [comment]
Yes, I can see, but in structure pruning, if I pruning channels in one layer, the input channels of consecutive layer also need to be changed. I don't think I can change it through yaml file. It's better to use loop way to build a model from scratch.

**Comment by glenn-jocher** [comment]
To modify the model structure for pruning, you'll need to adjust the PyTorch model code directly rather than using the YAML file. This allows you to handle dependencies between layers effectively.

**Comment by Christian-lyc** [comment]
Yes, thank you. Do you also provide pytorch model code that can load the model ultralytics yolov8n pretrained model? Thank you

**Comment by glenn-jocher** [comment]
The YOLOv8 models are built using PyTorch, and you can load the pretrained model using the Ultralytics package with `model = YOLO('yolov8n.pt')`. For custom modifications, you'll need to adjust the PyTorch code directly.
